packages/router/src/page-loader.tsx

```diff
@@ -128,14 +128,14 @@ export class PageLoader extends React.Component<Props> {
     // Remove the timeout because the page has loaded.
     this.clearLoadingTimeout()
 
+    this.context.setParams(props.params)
```

Comment:
I'm curious if this fully fixes the problem. What if you have `useParams()` in a layout outside of the route, in a `<Set>`? Do you still get a valid value for `params.documentId` for the first render?

Reply:
I'll test that now.

Reply:
It does indeed.

Reply:
Actually... it does not in tests, but it does when I actually used it. Wtf.

Reply:
Ah, false positive because of HMR.